tests/playwright/src/model/pages/create-machine-page.ts

```diff
@@ -0,0 +1,78 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Locator, Page } from '@playwright/test';
+import { expect as playExpect } from '@playwright/test';
+
+import { BasePage } from './base-page';
+import { ResourcesPage } from './resources-page';
+
+export class CreateMachinePage extends BasePage {
+  readonly heading: Locator;
+  readonly machineNameBox: Locator;
+  readonly podmanMachineConfiguration: Locator;
+  readonly imagePathBox: Locator;
+  readonly browseImagesButton: Locator;
+  readonly rootPriviledgesCheckbox: Locator;
+  readonly userModeNetworkingCheckbox: Locator;
+  readonly startNowCheckbox: Locator;
+  readonly closeButton: Locator;
+  readonly createMachineButton: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.heading = this.page.getByRole('heading', { name: 'Create Podman Machine' });
+    this.podmanMachineConfiguration = this.page.getByRole('form', { name: 'Properties Information' });
+
+    this.machineNameBox = this.podmanMachineConfiguration.getByRole('textbox', { name: 'Name' });
+    this.imagePathBox = this.page.getByRole('textbox', { name: 'Image Path (Optional) ' });
+    this.browseImagesButton = this.page.getByRole('button', { name: 'button-Image Path (Optional)' });
+    this.rootPriviledgesCheckbox = this.podmanMachineConfiguration
+      .getByRole('checkbox', { name: 'Machine with root privileges' })
+      .locator('..');
+    this.userModeNetworkingCheckbox = this.page.getByRole('checkbox', {
+      name: 'User mode networking (traffic relayed by a user process). See [documentation](https://docs.podman.io/en/latest/markdown/podman-machine-init.1.html#user-mode-networking).',
+    });
+    this.startNowCheckbox = this.page.getByRole('checkbox', { name: 'Start the machine now' });
+    this.closeButton = this.page.getByRole('button', { name: 'Close' });
+    this.createMachineButton = this.page.getByRole('button', { name: 'Create' });
+  }
+
+  async createMachine(machineName: string, isRootless: boolean): Promise<ResourcesPage> {
+    await playExpect(this.podmanMachineConfiguration).toBeVisible();
+    await this.machineNameBox.fill(machineName);
+
+    if (!isRootless) {
+      await playExpect(this.rootPriviledgesCheckbox).toBeVisible();
+      const upperElement = this.rootPriviledgesCheckbox.locator('..');
+      const clickableCheckbox = upperElement.getByText('Enable');
+      await clickableCheckbox.click();
+    }
+
+    await this.createMachineButton.click();
+
+    const successfulCreationMessage = this.page.getByText('Successful operation');
+    const goBackToResourcesButton = this.page.getByRole('button', { name: 'Go back to resources' });
+
+    await playExpect(successfulCreationMessage).toBeVisible({ timeout: 100000 });
```

Comment:
I would say that we need to improve the handling of what might happen once you click on Create button. 
we might got into what is here in case of success. What if there is an error? We should fail the test and throw that error that might appear in the form, right? Maybe also open log?

Reply:
Also, if there will be existing machine, this new one gonna get a dialogs regarding switching of the default connection. So it is going to be important to consider that aspect as well.